src/carma_clock.cpp

```diff
@@ -61,7 +61,8 @@ void CarmaClock::update(timeStampMilliseconds current_time) {
         // if not initialized then do it and let anyone waiting know
         _is_initialized = true;
         std::unique_lock lock(_initialization_mutex);
-        _initialization_cv.notify_one();
+        // Notify all blocked threads
+        _initialization_cv.notify_all();
```

Comment:
Why do we choose notify all over one? Is there a scenario that we are testing require it to notify all upon an update?

Reply:
the conditional variable here can be used to block one our many threads with the wait call. Notify One will only notify a single thread. If multiple threads are waiting on initialization, only 1 will be notified here. The desired behavior is to notify and unblock all threads as shown in my unit test. (https://en.cppreference.com/w/cpp/thread/condition_variable)